packages/common/src/index-options.ts

```diff
@@ -4,13 +4,7 @@ import {Replace} from './helpers/replace-type';
 export interface GSIIndexOptions {
   type: INDEX_TYPE.GSI;
   partitionKey: string;
-  sortKey: string;
-  /**
-   * DynamoDB only adds item to index if both Partition Key and Sort Key defined
-   * for any given GSI contains value
-   *
-   * Indexes missing sort key value will not be added to index.
-   */
+  sortKey?: string;
   isSparse?: boolean;
```

Comment:
Instead of removing this note, we should try to reword it, as it could be useful to folks not fully across the Sparse behaviour.
with something like this...

```typescript
/**
* When a Global Secondary Index (GSI) is created with both a partition key and a sort key, any items added * to the table that lack a partition key or sort key value will be excluded from the index.
* For GSIs that only have a partition key, the same behaviour applies, but only to the partition key.
* 
* In these scenarios `isSparse` should be set to True.
*/

```

Reply:
Added the comment but as a definition level jsdoc